src/lightmap.cpp

```diff
@@ -492,6 +492,46 @@ bool map::pl_line_of_sight( const tripoint &t, const int max_range ) const
 
 #include "messages.h"
 
+typedef ShadowcastingSlope slope;
+
+slope::ShadowcastingSlope( int x, int y )
```

Comment:
That looks like a constructor (and actually is one), but it has a strange name. Why are you using different type names here? It should be either `slope::slope` or `ShadowcastingSlope::ShadowcastingSlope`.